app/controllers/publishers/vacancies/application_forms_controller.rb

```diff
@@ -53,17 +53,24 @@ def update_vacancy
     if form.valid?
       vacancy.update(form.params_to_save)
       update_google_index(vacancy) if vacancy.listed?
-
-      if session[:current_step] == :review
-        redirect_updated_job_with_message
-      else
-        redirect_to organisation_job_build_path(vacancy.id, :documents)
-      end
+      redirect_to_next_step
     else
       render "publishers/vacancies/build/applying_for_the_job_details"
     end
   end
 
+  def redirect_to_next_step
+    return redirect_to organisation_job_path(vacancy.id) if save_and_finish_later?
+
+    if vacancy.completed_steps.exclude?("documents")
```

Comment:
As you added this here, would it be possible to remove this from `#next_invalid_step`?

https://github.com/DFE-Digital/teaching-vacancies/blob/fadd7a813255cd54b6c410a239c0a5b4e3a48404/app/controllers/publishers/vacancies/base_controller.rb#L46

Reply:
Or actually would the  opposite be better? As in to move this check inside `#next_invalid_step` 

Thoughts?

Reply:
Take a look at the latest commit I've pushed. Hopefully, this should make it all a bit clearer/nicer.